app/models/grade_entry_item.rb

```diff
@@ -5,7 +5,7 @@ class GradeEntryItem < ApplicationRecord
 
   has_one :course, through: :grade_entry_form
 
-  has_many :grades, dependent: :delete_all
+  has_many :grades, dependent: :destroy
```

Comment:
Okay I realized after reviewing again that this change will also change what happens when the user tries to destroy a single column (on the grade entry form "update" page). Previously, we actually did allow the user to delete the column even if that would result in non-nil grades being deleted.

I don't want to change that existing behaviour in this PR. So, please revert the added `Grade` destroy callback and this line, and instead add a callback to `GradeEntryForm` itself to check for non-nil grades. See also my comment on the form below about how I want you to check for this.

Sorry for the change of plans!